frontend/website/docs/faq.md

```diff
@@ -66,3 +66,8 @@ Coda does not support smart contracts currently. The development team is looking
 
 No, Coda does not natively implement privacy features at the moment. However, privacy is key consideration for cryptocurrencies, and is also on the development roadmap.
 
+#### Does Coda solve the data availability problem?
+
+The data availability problem, while a concern for other blockchains, does not apply to the Coda network. This is because there is no concept of a "light client" or SPV in Coda. Every node is a full node, and fully verifies the state of the blockchain in a lightweight manner by verifying a zk-SNARK.
+
+In other networks, malicious block producers can send just the block header containing hashes and merkle roots for a new block to light clients. They can then collude to prevent other full nodes from receiving the rest of the block data (hence the name data availability problem). This forces the other block producers to either stall the network, or revert back to a prior block, causing issues for all the light clients that believed the new state. In Coda, because every node can independently verify state, this divide and conquer strategy cannot work.
```

Comment:
This answer is wrong, and the attack *does* work, albeit only with a malicious fork long enough that the underlying transactions are no longer available. The attacks that apply to Coda involve the adversary producing a *valid* chain and declining to provide the underlying state, which snarks don't help with at all. The issue Vitalik talks about with producing fraud proofs for *invalid* blocks is only one instance of the general data availability problem.

Mallory produces a chain of valid blocks offline, long enough to be selected over the honest chain (by assumption), and with her portion longer than *k* blocks. She then joins the network and publishes her best tip. Newly joining clients attempt to bootstrap, and since Mallory's best tip beats the honest one they try to sync to her transition frontier root's snarked ledger. She refuses to service sync requests, bootstrap never completes, and the network is bricked from the perspective of new nodes. If instead the fork point of Mallory's chain is <= *k* blocks ago, bootstrap succeeds since honest nodes have the snarked ledger at that point, but eager catchup fails when she refuses to services requests to fetch the blocks between the root and her best tip. Normal catchup to the honest chain will still work when an honest block is received from gossip though. If the fork point is >= *2k + Δ*, bootstrap will be triggered on old nodes as well as new ones and everyone will be bricked, no new blocks will be produced at all.

These scenarios rely on Mallory being able to produce malicious chains of moderately high lengths, so any security argument will rely on the difficulty of doing so under our honest majority modelling assumption.

Reply:
@enolan I was generalizing over the data availability problem as experienced by other networks due to the existence of light clients. Generally, that's what most folks have as their concern when a FAQ about data availability attacks comes up.

I think you're highlighting an attack that is analogous, and is related to having incomplete chain data, but I wouldn't call this the same data availability problem as commonly understood. Happy to include another question / answer addressing this caveat if you think this is incomplete, but would want to probe into the security assumptions of this long fork attack.

Open questions on my end:
- I thought we had a mechanism to combat the long fork attack without relying on chain history?
- seems like the data availability problem via light clients is an easier attack since it requires only one block to be able to execute (provided you have a dishonest majority). This attack doesn't need a majority, but rather the ability to build up chain strength faster than the honest network. What's the effort involved to do this, and wouldn't an honest node just be able to verify against several peers, and blacklist the one that is giving a different snarked ledger?
- what is the likelihood of / how would colluding block producers being able to secretly build upon a covert chain and then expose the chain when the fork points is greater than 2k + delta? would this require an alternate gossip channel, presumably because otherwise each BP would have to construct a separate long fork offline?
- I assume the existence of archive nodes that the network could query upon observation of a long fork attack would resolve this issue quickly? albeit at a cost of centralization. Would the adversaries be able to capture archive nodes, in order to brick the network per your scenario?

Thanks for explaining this, I can add a section about this if we feel it's necessary... but generally the FAQ is a high level doc